geoapps/inversion/params.py

```diff
@@ -346,6 +347,16 @@ def mesh(self):
     @mesh.setter
     def mesh(self, val):
         self.setter_validator("mesh", val, fun=self._uuid_promoter)
+
+        if (
```

Comment:
Ah, ok - so this is why you removed all the data rotation stuff!

Reply:
Same comment as above, looks like we were already blocking the process down the road. I am just formalizing it here. We would need to make serious push on all inversion methods to handle the rotation properly.